src/Supervisor.cpp

```diff
@@ -867,3 +954,47 @@ u16 GetInput(void)
 
     return GetControllerInput(buttons);
 }
+
+#pragma optimize("s", on)
+#pragma var_order(diprange, result)
+BOOL CALLBACK controllerCallback(LPCDIDEVICEOBJECTINSTANCEA lpddoi, LPVOID pvRef)
+
+{
+    HRESULT result;
+    DIPROPRANGE diprange;
+    result = (HRESULT)pvRef; // ehm, ok?
```

Comment:
Maybe drop the cast, retype `result` to LPVOID, and rename it to something like pvRefBackup or something? IMO it'd make a bit more sense.

Reply:
Makes sense, ghidra's output led me to believe at first that it was the result of `g_Supervisor.controller->SetProperty`